components/cli/pkg/registry/registry.go

```diff
@@ -0,0 +1,181 @@
+	Push(parsedCellImage *image.CellImage, fileBytes []byte, username, password string) error
+	Out() io.Writer
+}
+
+type CelleryRegistry struct {
+	hub *registry2.Registry
+}
+
+func NewCelleryRegistry() *CelleryRegistry {
+	registry := &CelleryRegistry{}
+	return registry
+}
+
+func (registry *CelleryRegistry) Push(parsedCellImage *image.CellImage, fileBytes []byte, username, password string) error {
+	fmt.Println("\nConnecting to ", util.Bold(parsedCellImage.Registry))
+	// Initiating a connection to Cellery Registry
+	hub, err := registry2.New("https://"+parsedCellImage.Registry, username, password)
+	if err != nil {
+		return fmt.Errorf("failed to initialize connection to Cellery Registry %v", err)
+	}
+	imageName := fmt.Sprintf("%s/%s:%s", parsedCellImage.Organization, parsedCellImage.ImageName,
+		parsedCellImage.ImageVersion)
+	repository := parsedCellImage.Organization + "/" + parsedCellImage.ImageName
+	// Creating the Cell Image Digest (Docker file Layer digest)
+	hash := sha256.New()
+	hash.Write(fileBytes)
+	sha256sum := hex.EncodeToString(hash.Sum(nil))
+	cellImageDigest := digest.NewDigestFromEncoded(digest.SHA256, sha256sum)
+
+	// Checking if the the Cell Image already exists in the registry
+	fmt.Println(fmt.Sprintf("\nChecking if the image %s already exists in the Registry", util.Bold(imageName)))
+	cellImageDigestExists, err := hub.HasBlob(repository, cellImageDigest)
+	if err != nil {
+		return err
+	}
+
+	// Pushing the cell image if it is not already uploaded
+	if !cellImageDigestExists {
+		fmt.Println(fmt.Sprintf("Pushing image %s", util.Bold(imageName)))
+		// Read stream of files
+		err = hub.UploadBlob(repository, cellImageDigest, bytes.NewReader(fileBytes), nil)
+		if err != nil {
+			return err
+		}
+	} else {
+		fmt.Println(fmt.Sprintf("\nUsing already existing image %s in %s Registry", util.Bold(imageName),
+			util.Bold(parsedCellImage.Registry)))
+	}
+
+	// Creating a Docker manifest to be uploaded
+	cellImageManifest := &schema1.Manifest{
+		Name: repository,
+		Versioned: manifest.Versioned{
+			SchemaVersion: 1,
+			MediaType:     schema1.MediaTypeSignedManifest,
+		},
+		Tag:          parsedCellImage.ImageVersion,
+		Architecture: "amd64",
+		FSLayers: []schema1.FSLayer{
+			{BlobSum: cellImageDigest},
+		},
+		History: []schema1.History{
+			{},
+		},
+	}
+	// Signing the Docker Manifest
+	key, err := libtrust.GenerateECP256PrivateKey()
+	if err != nil {
+		return fmt.Errorf("error occurred while pushing the cell image, %v", err)
+	}
+	signedCellImageManifest, err := schema1.Sign(cellImageManifest, key)
+	if err != nil {
+		return fmt.Errorf("error occurred while pushing the cell image, %v", err)
+	}
+
+	// Uploading the manifest to the Cellery Registry (Docker Registry)
+	err = hub.PutManifest(repository, parsedCellImage.ImageVersion, signedCellImageManifest)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("\nImage Digest : %s\n", util.Bold(cellImageDigest))
```

Comment:
Use the stdout provided by Out() function in registry interface to print.

Reply:
Updated.